### PR TITLE
Form import: map only items visible in the current entity

### PR DESCRIPTION
**Pull request by AdrienClairembault**
The form import process try to match items using their names.
It should be restricted to the current entity to make sure the user don't see data that he isn't allowed to see.

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets |

### CODE REVIEW

**Review by cedric-anne — COMMENTED** [review]
I am not sure that it is a good idea to rely on session for a such low-level method. Do you think it could be possible to pass some restriction parameters to the `DatabaseMapper` when it is instanciated, instead of relying on the session ? For instance, `getEntitiesRestrictCriteria` can be call with a specific list of entities to accept.

**Comment by AdrienClairembault** [comment]
Good idea, I'll look into it.

**Comment by AdrienClairembault** [comment]
Rebase to main to fix conflicts.

**Comment by AdrienClairembault** [comment]
Be careful when merging; it will probably impact #17670 (whichever get merged first, the other one will need a rebase).

File: src/Glpi/Form/Export/Context/DatabaseMapper.php [hunk 1]
```diff
@@ -124,4 +138,57 @@ private function contextExist(string $itemtype, string $name): bool
...
+        ];
+        $condition = [
+            'name' => $name
+        ];
+
+        // Compute default where (visibility restrictions)
+        $default_where = Search::addDefaultWhere($itemtype);
+        if ($default_where !== '') {
+            $condition[] = new QueryExpression($default_where);
+        }
```
[the remark this answers is not in the document]
Comment (AdrienClairembault) [reply src/Glpi/Form/Export/Context/DatabaseMapper.php:164]: That seems reasonable, I'll do it.

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**